File: python/test/jaccardDistanceTest.py
import unittest
from jaccardDistance import calculate

class TestJaccardDistance(unittest.TestCase):

  def test_identical_words(self):
      index = calculate("transport","transport")
      self.assertEqual(index, 0)

  def test_words_with_no_similarity(self):
      index = calculate("aaa","bbb")
      self.assertEqual(index, 1)

  def test_words_with_similarity(self):
      index = calculate("transport","transportation")
      self.assertEqual(index, 0.125)

File: java/src/test/java/com/hakanozbay/metric/JaccardDistanceTest.java
package com.hakanozbay.metric;

import static org.junit.Assert.assertEquals;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class JaccardDistanceTest {

	JaccardDistance jaccardDistance;
	
	@Before
	public void setUp()
	{
		jaccardDistance = new JaccardDistance();
	}
	
	@After
	public void taerDown()
	{
		jaccardDistance = null;
	}
	
	@Test
	public void testIdenticalWords()
	{
		double distance = jaccardDistance.calculate("transport", "transport");
		assertEquals(0, distance, 0);
	}

	@Test
	public void testWordsWithNoSimilarity()
	{
		double distance = jaccardDistance.calculate("aaa", "bbb");
		assertEquals(1, distance, 0);
	}
	
	@Test
	public void testWordsWithSimilarity()
	{
		double distance = jaccardDistance.calculate("transport","transportation");
		assertEquals(0.125, distance, 0);
	}

	  
}
